pkg/pvc/pvc_protected_entity.go

```diff
@@ -326,6 +287,57 @@ func (this PVCProtectedEntity) getProtectedEntityForPV(ctx context.Context, pv *
+	if err != nil {
+		errorMsg := fmt.Sprintf("Could not decode snapshot ID encoded string %s", snapshotID64Str)
+		this.logger.WithError(err).Error(errorMsg)
+		return []astrolabe.ProtectedEntity{}, errors.Wrap(err, errorMsg)
+	}
+	decodedPEID, err := astrolabe.NewProtectedEntityIDFromString(string(snapshotIDBytes))
+	if err != nil {
+		errorMsg := fmt.Sprintf("Could not translate decoded snapshotID %s into pe-id", string(snapshotIDBytes))
+		this.logger.WithError(err).Error(errorMsg)
+		return []astrolabe.ProtectedEntity{}, errors.Wrap(err, errorMsg)
+	}
+	// Determine the component pe-type, should be either ivd or para-virt
+	expectedComponentPEType := this.getComponentPEType()
+	if expectedComponentPEType == decodedPEID.GetPeType() {
+		// If the decoded snap is of the expected component pe-type return.
+		componentPE, err := this.ppetm.pem.GetProtectedEntity(ctx, decodedPEID)
+		if err != nil {
+			return []astrolabe.ProtectedEntity{}, errors.Wrapf(err, "Could not get Protected Entity for PE %s", decodedPEID.String())
+		}
+		this.logger.Infof("Retrieved the component PE %s from the snapshot pe-id: %s", componentPE.GetID().String(), snapshotToDelete.String())
+		return []astrolabe.ProtectedEntity{componentPE}, nil
+	}
+	if expectedComponentPEType == astrolabe.IvdPEType && decodedPEID.GetPeType() == astrolabe.ParaVirtPvPEType {
+		// If the decoded snap is not of the expected component pe-type, decode further.
+		// This scenario occurs when backup is created in Guest and delete is attempted in Vanilla/Supervisor.
+		this.logger.Infof("Further decoding the snapshot pe-id %s from the original snapshot pe-id", decodedPEID.GetSnapshotID().String(), snapshotToDelete.String())
+		return this.getComponentsFromSnapshot(ctx, decodedPEID.GetSnapshotID())
+	} else if expectedComponentPEType == astrolabe.ParaVirtPvPEType && decodedPEID.GetPeType() == astrolabe.IvdPEType {
+		// The decoded snap-id component type is of ivd when the expected type is supposed to be ivd.
+		// The scenario occurs when backup is created in Vanilla/Supervisor and delete is attempted in Guest.
+		// Re-encode the pe-id into a para-virt type.
+		encodedSnapshotStr := base64.RawStdEncoding.EncodeToString([]byte(decodedPEID.String()))
+		componentSnapshotId := astrolabe.NewProtectedEntitySnapshotID(encodedSnapshotStr)
+		encodedPeId := astrolabe.NewProtectedEntityIDWithSnapshotID(astrolabe.ParaVirtPvPEType, "ivd-"+decodedPEID.GetID(), componentSnapshotId)
+		encodedComponentPE, err := this.ppetm.pem.GetProtectedEntity(ctx, encodedPeId)
+		if err != nil {
+			return []astrolabe.ProtectedEntity{}, errors.Wrapf(err, "Could not get Protected Entity for PE %s", encodedPeId.String())
+		}
+		return []astrolabe.ProtectedEntity{encodedComponentPE}, nil
+	} else {
+		errMsg := fmt.Sprintf("Unsupported PE type: %s when expected PE type: %s for pe-id: %s", decodedPEID.GetPeType(), expectedComponentPEType, decodedPEID.String())
+		this.logger.Errorf(errMsg)
```

Comment:
Errorf --> Error

Reply:
Actually, `errMsg` has formatting, so it has to be Errorf.